src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -6,26 +6,60 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel
 {
+    /// <summary>
+    /// Provides programmatic configuration of Kestrel-specific features.
+    /// </summary>
     public class KestrelServerOptions
     {
+        /// <summary>
+        /// Enables the UseKestrel options callback to resolve and use services registered by the application during startup.
+        /// Typically initialized by <see cref="Hosting.WebHostBuilderKestrelExtensions.UseKestrel(Hosting.IWebHostBuilder, Action{KestrelServerOptions})"/>.
+        /// </summary>
         public IServiceProvider ApplicationServices { get; set; }
 
+        /// <summary>
+        /// Gets or sets an <see cref="IConnectionFilter"/> that allows each connection <see cref="System.IO.Stream"/>
+        /// to be intercepted and transformed.
+        /// Configured by the <c>UseHttps()</c> and <see cref="Hosting.KestrelServerOptionsConnectionLoggingExtensions.UseConnectionLogging(KestrelServerOptions)"/>
+        /// extension methods.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to null.
+        /// </remarks>
         public IConnectionFilter ConnectionFilter { get; set; }
 
+        /// <summary>
+        /// Equivalent to <see cref="System.Net.Sockets.Socket.NoDelay"/>. Set to false to enable Nagle's algorithm.
```

Comment:
Yeah, it's fine on it's own.  The polarity is a little weird, but that's not the docs fault.